src/Stache/Repositories/CollectionRepository.php

```diff
@@ -99,4 +100,20 @@ public static function bindings(): array
             Collection::class => \Statamic\Entries\Collection::class,
         ];
     }
+
+    public function addExtraPreviewTargets($handle, $targets)
+    {
+        $targets = collect($this->extraPreviewTargets[$handle] ?? [])
+            ->merge($targets)
+            ->unique(function ($target) {
+                return $target['format'];
+            })->all();
+
+        $this->extraPreviewTargets = array_merge($this->extraPreviewTargets, [$handle => $targets]);
+    }
+
+    public function extraPreviewTargets($handle)
+    {
+        return collect($this->extraPreviewTargets[$handle] ?? []);
+    }
```

Comment:
Should we combine these methods to simply `extraPreviewTargets`. Then it would be coherent with the `extraPreviewTargets` method in the `Collection` class.

Reply:
Nah I think the multiple arguments throws it off a bit here. I'm fine with having two methods in this class.